src/main/java/io/liftbridge/Client.java

```diff
@@ -1,55 +1,79 @@
 package io.liftbridge;
 
 import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
 import io.grpc.stub.StreamObserver;
-import io.liftbridge.MessageHandler;
 import io.liftbridge.proto.APIGrpc.APIStub;
 import io.liftbridge.proto.APIGrpc.APIBlockingStub;
 import io.liftbridge.proto.APIGrpc;
 import io.liftbridge.proto.Api;
 import com.google.protobuf.ByteString;
-import java.nio.ByteBuffer;
+
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+/**
+ * {@code Client} is the main API used to communicate with a Liftbridge cluster. Use {@link Builder#build()} to
+ * construct a {@code Client} instance.
+ */
 public class Client {
+
     private APIStub asyncStub;
     private APIBlockingStub blockingStub;
-    private Client(APIStub asyncStub, APIBlockingStub blockingStub) {
-        this.asyncStub = asyncStub;
-        this.blockingStub = blockingStub;
+    private ManagedChannel channel;
+
+    private Client(ManagedChannel channel) {
+        this.channel = channel;
+        this.asyncStub = APIGrpc.newStub(channel);
+        this.blockingStub = APIGrpc.newBlockingStub(channel);
     }
 
     /**
-     * Creates a client connected to the API using the provided gRPC channel.
+     * Closes the client connection by shutting down the gRPC channel and releasing any resources.
      */
-    public static Client connect(ManagedChannel grpcChannel) {
-        return new Client(APIGrpc.newStub(grpcChannel),
-                          APIGrpc.newBlockingStub(grpcChannel));
+    public void close() {
+        channel.shutdown();
     }
 
-
     /**
      * Creates a stream. Will block until response is received.
      */
     public void createStream(String streamName, StreamOptions opts) {
         blockingStub.createStream(opts.asRequest(streamName));
     }
 
+    /**
+     * Creates a stream. Will block until response is received.
+     */
+    public void createStream(String streamName) {
+        createStream(streamName, new StreamOptions());
+    }
+
     /**
      * Subscribes to a stream.
      */
-    public void subscribe(String streamName, MessageHandler msgHandler,
-                          SubscriptionOptions opts) {
+    public void subscribe(String streamName, SubscriptionOptions opts, MessageHandler msgHandler) {
```

Comment:
Making the `MessageHandler` the last argument feels a bit more ergonomic since it's typically implemented anonymously inline.